lib/ansible/modules/cloud/vmware/vmware_guest_attribute.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright, (c) 2018, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: vmware_guest_attribute
+short_description: Manage custom attributes from VMWare for virtual machine
+description:
+    - This module can be used to add, remove, list custom attributes for the given virtual machine.
+version_added: 2.6
+author:
+    - Jimmy Conner
+    - Abhijeet Kasurde (@akasurde) <akasurde@redhat.com>
+notes:
+    - Tested on vSphere 6.5
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+   name:
+     description:
+     - Name of the virtual machine to work with.
+     required: True
+   state:
+     description:
+     - The action to take.
+     default: 'list'
+     choices: ['list', 'present', 'absent']
+   uuid:
+     description:
+     - UUID of the instance to manage if known, this is VMware's unique identifier.
+     - This is required parameter, if C(name) is not supplied.
+   folder:
+     description:
+     - Destination folder, absolute path to find an existing guest.
+     - This is required parameter, if C(name) is supplied and multiple virtual machines with same name are found.
+   datacenter:
+     description:
+     - Destination datacenter for the deploy operation
+     required: True
+   attribute_name:
```

Comment:
I'd suggest this to be a list, so you can add multiple key/values in a single go (might rename it to `vmware_guest_custom_attributes`)
```
- name: Add custom values
  vmware_guest_custom_attributes:
    hostname ...
    ...
    attributes:
    - name: Key1
      value: Value1
    - name: Key2
      value: Value2
```

A loop causes the whole python script to be executed multiple times, which is much slower than a single api call.